proposals/NNNN-android-oreo-foreground-services.md

```diff
@@ -0,0 +1,145 @@
+
+* Proposal: [SDL-NNNN](NNNN-android-oreo-foreground-services.md)
+* Author: [Kujtim Shala](https://github.com/kshala-ford)
+* Status: **Awaiting review**
+* Impacted Platforms: [ Android ]
+
+## Introduction
+
+This proposal addresses an issue with Android Oreo background requirements forcing SDL for Android become potentially intrusive to all users with or without owning an SDL capable vehicle.
+
+## Motivation
+
+With Android Oreo Google added [Background Service Limitations](https://developer.android.com/about/versions/oreo/background#services) for services running in the background.  Services are used to maintain SDL proxies as well as the router service providing multiplexing. With [SDL Android v4.5.0](https://github.com/smartdevicelink/sdl_android/releases/tag/4.5.0) a fix was added to make the router service visible using a notification shown in the Android notification center. This notification is visible for 10 to 20 seconds (depending on the number of SDL apps installed on the phone).
+
+![screenshot](../assets/proposals/NNNN-android-oero-foreground-services/sdl_notification.jpg)
+
+[SDL Android v4.6.0](https://github.com/smartdevicelink/sdl_android/tree/release/4.6.0_RC) (currently under development) improves the behavior adding flags to known bluetooth devices. This flag will indicate if the app ever connected and registered to the connected Bluetooth device. 
+
+> The new logic will keep the service in the foreground for 10 seconds if it is a device that has never been seen before or rather the first time the device has connected over bluetooth, if the device does NOT connect, the mac address of the device will be stored with a flag indicating it should be ignored. If it DOES connect over SDL, the notification will then be present for 20 seconds on connection to that device to ensure a connection happens in all scenarios. The router service will still be started for devices that have the flag indicating to ignore those devices, but the service is immediately taken out of the foreground state. This is necessary since in Android O+ services have to be started into the foreground when started by background applications.
+
+This behavior is way better compared to v4.5.0 however the SDL related notification will still be visible to pure non-SDL users. We expect the the risk app partners may weigh benefits integrating SDL and complains from non-SDL users potentially giving bad rates or stop using the app.
+
+## Proposed solution
+
+The proposed solution is to change the behavior how apps prepare the SDL connection to Bluetooth devices. The idea is to create the SDL proxy only while the app is in foreground on the phone in case it never connected to a head unit. It's not allowed to start the router service from the background if the connecting BT device is not known that it supports SDL. Once the app has registered to a head unit it's allowed to start the router service in future connections with that BT device.

```

Comment:
Please update to "...change the behavior of how apps prepare..." and "...in case it never connects to a head unit."